TECHMANIA/Assets/Scripts/UniversalIO.cs

```diff
@@ -45,4 +149,40 @@ public static string ReadAllText(string path)
             return File.ReadAllText(path);
         }
     }
+
+    public static void WriteAllText(string path, string content)
+    {
+        if (path.StartsWith(ANDROID_CONTENT_URI))
+        {
+            AndroidNativeIO.IO.File.WriteAllText(path, content);
+        } else
+        {
+            File.WriteAllText(path, content);
+        }
+    }
+
+    #endregion
+
+    #region Path
+
+    public static string PathGetDirectoryName(string path, bool returnToTreeUri = true)
```

Comment:
Also, it was my oversight that there are 2 ways to get the directory name (`DirectoryInfo.Name` and `Path.GetDirectoryName`). Could you test if it's okay to replace the former with latter?